src/xsar/sentinel1_dataset.py

```diff
@@ -126,8 +124,16 @@ def __init__(self, dataset_id, resolution=None,
                         'xtrack': self._dataset.digital_number.xtrack}
             )
 
-        self._dataset.attrs.update(self.s1meta.to_dict("all"))
+        # FIXME possible memory leak
+        # when calling a self.s1meta method, an ActorFuture is returned.
+        # But this seems to break __del__ methods from both Sentinel1Meta and XmlParser
+        # Is it a memory leak ?
+        # see https://github.com/dask/distributed/issues/5610
+        # tmp_f = self.s1meta.to_dict("all")
+        # del tmp_f
+        # return
 
+        self._dataset.attrs.update(self.s1meta.to_dict("all"))
         # load subswath geometry
         # self.geometry = self.load_geometry(self.files['noise'].iloc[0])
         # self._dataset.attrs['geometry'] = self.geometry
```

Comment:
Removing comment line

Reply:
ok, some old stuff here. can be removed.